public/app/plugins/panel/alertGroups/AlertmanagerPicker.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useMemo } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { Select } from '@grafana/ui';
+import { AlertManagerDataSource, GRAFANA_RULES_SOURCE_NAME } from 'app/features/alerting/unified/utils/datasource';
+
+interface Props {
+  onChange: (alertManagerSourceName: string) => void;
+  current?: string;
+  dataSources: AlertManagerDataSource[];
+}
+
+function getAlertManagerLabel(alertManager: AlertManagerDataSource) {
+  return alertManager.name === GRAFANA_RULES_SOURCE_NAME ? 'Grafana' : alertManager.name.slice(0, 37);
+}
+
+export const AlertManagerPicker = ({ onChange, current, dataSources }: Props) => {
```

Comment:
I'm not sure if I understand the question :D I've copied most of the original code of the `AlertManagerPicker` here since this seems to be the only place in Grafana where we allow the user to select an Alertmanager source from a dropdown where we can't wrap the parent in our new `Context`.